User.php

```diff
@@ -0,0 +1,271 @@
+    $pattern = "/^[a-zA-Z ]{1,25}$/";
+
+    // Error in first name or last name returns the first name or last name 
+    // respectively.
+    if (!preg_match($pattern, $this->fname))
+      return $this->fname;
+    else if (!preg_match($pattern, $this->lname))
+      return $this->lname;
+    else {
+      // On valid name, the first and last name are concatenated and returned.
+      $greetings = $this->fname . " " . $this->lname;
+      return $greetings;
+    }
+  }
+
+  /**
+   * The function stores the image submitted by user in 'uploads' folder.
+   * 
+   * @return string
+   *  Returns the target address of stored image on successful upload.
+   */
+  public function isValidImage(): string {
+    // Storing the target directory name.
+    $targetDir = "./uploads/";
+    // Contatenating  the target directory to the image file name to get the 
+    // full path where the image will be stored.
+    $target = $targetDir . basename($_FILES['image']['name']);
+    // Storing the temporary file name of the image.
+    $tempFileName = $_FILES['image']['tmp_name'];
+    $imageFileType = strtolower(pathinfo($target, PATHINFO_EXTENSION));
+    // Checking for correct image extension type.
+    if ($imageFileType != "jpg" && $imageFileType != "png" && $imageFileType != "jpeg" && $imageFileType != "gif") {
+      return "";
+    }
+    // The path is returned if the image is uploaded successfully.
+    if (move_uploaded_file($tempFileName, $target)) {
+      return $target;
+    }
+    // Returning null on failure in image upload.
+    else
+      return "";
+  }
+
+  /**
+   * Checks if the number is valid or not.
+   * The number must be an Indian mobile number.
+   * 
+   * @return bool|string
+   *  Returns 1 in case of invalid number and the number as string if valid.
+   */
+  public function isValidNumber(): mixed {
+    // Checking for proper mobile number pattern.
+    if (!preg_match("/^(\+91)[0-9]{10}$/", $this->mobNo)) {
+      return 1;
+    }
+    else {
+      return $this->mobNo;
+    }
+  }
+
+  /**
+   * Checks the validity of the email syntax and also if the email is correct.
+   * 
+   * @return bool 
+   *  Returns 0 on invalid syntax and 1 on correct email.
+   */
+  public function isValidEmail(): bool {
+    // If email syntax is incorrect, returns 0.
+    if (!filter_var($this->email, FILTER_VALIDATE_EMAIL)) {
+      return 0;
+    }
+    else {
+      // Stores the api key.
+      $api_key = "038c427c27d1417397129856c0f90f04";
+      // Initialising curl session.
+      $ch = curl_init();
+      // Settings the options to include when connecting with the api.
+      curl_setopt_array($ch, [
+        CURLOPT_URL => "https://emailvalidation.abstractapi.com/v1/?api_key=$api_key&email=$this->email",
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_FOLLOWLOCATION => true
+      ]);
+      // Establishing connection with the api using the above option values.
+      $result = curl_exec($ch);
+      curl_close($ch);
```

Comment:
Why we are using crul not guzzle? We should use Guzzle.

Reply:
ok